util/progress/progressui/display.go

```diff
@@ -103,28 +103,31 @@ type displayInfo struct {
 }
 
 type job struct {
-	startTime     *time.Time
-	completedTime *time.Time
-	name          string
-	status        string
-	hasError      bool
-	isCanceled    bool
-	vertex        *vertex
-	showTerm      bool
+	intervals   []interval
+	isCompleted bool
+	name        string
+	status      string
+	hasError    bool
+	isCanceled  bool
+	vertex      *vertex
+	showTerm    bool
 }
 
 type trace struct {
 	w             io.Writer
+	startTime     *time.Time
 	localTimeDiff time.Duration
 	vertexes      []*vertex
 	byDigest      map[digest.Digest]*vertex
 	nextIndex     int
 	updates       map[digest.Digest]struct{}
 	modeConsole   bool
+	groups        map[string]*vertexGroup // group id -> group
```

Comment:
If all items would be "groups" and vertex without a progressgroup would just be a group with only one item I think some duplications could be avoided and maybe easier to understand. Not needed atm unless you think it is easy but I think worth considering as optimization.

We can assume that vertex digest and progress group id never collide.